rep-0151.rst

```diff
@@ -0,0 +1,260 @@
+``python3`` [7]_.
+The ``python`` command cannot be trusted to a spefic Python version.
+On older ROS distros, scripts can continue to use ``python`` since they're known
+to work on those platforms.
+In preparation for Noetic, these shebang's should be rewritten to the specific
+version of Python supported, ``python3``.
+
+Packages using the same branch will need to conditionally rewrite the shebangs.
+Packages can use the CMake macro ``catkin_install_python()`` to install Python
+scripts with rewritten shebangs.
+
+The same issue appears in scripts that call the ``python`` command directly.
+If they are Python scripts, they should be modified to invoke ``sys.executable``.
+Otherwise, they should be edited to invoke the specific version of Python they
+require, or ``python$ROS_PYTHON_VERSION`` if the script works with both.
+
+Dependencies and package.xml
+''''''''''''''''''''''''''''
+
+On platforms where the target version of Python is 2, the package.xml of a ROS
+package must refer to Python 2 dependencies, and when the target Python
+version is 3 it must refer to Python 3 dependencies.
+Packages which release from different branches for each ROS distro can replace
+rosdep keys that resolve to Python 2 dependencies to ones that resolve to
+Python 3 equivalents.
+Packages using the same code base for multiple ROS distros should instead use
+conditional dependencies as described in REP 149 [5]_.
+
+.. code-block: xml
+
+    <depend condition="$ROS_PYTHON_VERSION == '2'">python-numpy</depend>
+    <depend condition="$ROS_PYTHON_VERSION == '3'">python3-numpy</depend>
+
+If ``ROS_PYTHON_VERSION`` is relied upon at build time, such as when using
+``catkin_install_python()`` to rewrite shebangs, then the package must declare a
+``<buildtool_depend>`` on ``ros_environment``.
+Any ROS package which uses ``ROS_PYTHON_VERSION`` in a script intended to be
+run at runtime should add an ``<exec_depend>`` tag for ``ros_environment``.
+
+Making Python fixes available to downstream packages
+''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+Transitioning to Python 3 is expected to be a significant effort.
+Typicically ROS packages are tested using the ROS build farm; however, that
+will not be available until packages for the targeted Ubuntu distribution
+become available.
+Instead, a placehoder ``Noetic`` ``distribution.yaml`` will be made available in
+advance of the buildfarm availability.
+Maintainers should add ``source`` entries for their Noetic branches to this file
+to enable downstream users to use ``rosinstall_generator`` with the
+``--upstream-development`` flag to get Python 3 fixes.
+Instructions to build from source using Python 3 will be made available to
+the ROS community.
+
+Once the build farm is available, Maintainers should release packages with
+Python 3 fixes to Noetic as soon as possible, even if they intend to make
+breaking changes later.
+
+Organizing Community effort
```

Comment:
```suggestion
Organizing community effort
```
Is community usually capitalized in this context? Other subtitles aren't title case here so this is either a typo or special case

Reply:
Do you think these titles should be capitalized? I wasn't sure. I'll commit the change as a typo for now.

Reply:
The vast majority of docs across OR don't capitalize titles (they use sentence case). So I say stick with the trend. As long as it's consistent within this document.